### PR TITLE
Task 4

### PR DESCRIPTION
**Pull request by schweppes2205**

### CODE REVIEW

File: task_4/module/ec2/output.tf [hunk 1]
```diff
@@ -0,0 +1,19 @@
+output "instances_id" {
+  value = aws_instance.instance[*].id
+}
+
+# output "instances_private_ip" {
```
Comment (kravetsd) [inline task_4/module/ec2/output.tf]: Please remove unnecessary comments.

File: task_4/module/ec2/main.tf [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  root_block_device {
+    volume_type = var.root_volume_param.volume_type
+    volume_size = var.root_volume_param.volume_size
+  }
+  vpc_security_group_ids = var.sg_id_list
+  user_data              = file("${path.module}/userdata.sh")
+  key_name               = var.key_pair_name
+}
+
+resource "null_resource" "data_from_instance" {
```
Comment (kravetsd) [inline task_4/module/ec2/main.tf]: For Task4 only. 
Could you please move this extra step to the root module, Let's say it is not a part  of basic EC2 module.
After you move it to root module, please check if it works for all created instances.

File: task_4/module/ec2/output.tf (hunk 1 above)
Comment (schweppes2205) [reply task_4/module/ec2/output.tf]: done

File: task_4/module/ec2/main.tf (hunk 2 above)
Comment (schweppes2205) [reply task_4/module/ec2/main.tf]: new ec2 module output generates an array of ip addresses from the module execution
```terraform
output "instances_public_ip" {
  value = [for i in aws_instance.instance : i.public_ip]
}
```

root module uses it to create a local variable with all public ip addresses from all ec2 module executions
```terraform
locals {
  ec2_pub_ip = flatten([for k, v in module.ec2 : v.instances_public_ip])
}
```
and use it to launch null_resource
```terraform
resource "null_resource" "data_from_instance" {
  count = length(local.ec2_pub_ip)
  connection {
...
    host        = local.ec2_pub_ip[count.index]
...
```

File: task_4/module/ec2/main.tf [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  root_block_device {
+    volume_type = var.root_volume_param.volume_type
+    volume_size = var.root_volume_param.volume_size
+  }
+  vpc_security_group_ids = var.sg_id_list
+  user_data              = file("${path.module}/userdata.sh")
+  key_name               = var.key_pair_name
+}
+
+# resource "null_resource" "data_from_instance" {
```
Comment (kravetsd) [inline task_4/module/ec2/main.tf]: Don't forget to remove commented changes it they are not required.

**Pull request merged**